index.html

```diff
@@ -5132,6 +5132,12 @@ <h2>The <code>application/credential+ld+json</code> Media Type</h2>
     the use of a specific media type.
   </p>
 
+  <p>
+    This media type MUST NOT be used with <dfn>embedded proof</dfn>.
+  </p>
+  <p>
+    This media type MAY be used with <dfn>external proof</dfn>.
```

Comment:
```suggestion
    This media type is used with credentials secured using an <dfn>external proof</dfn>.
```

Suggest that we try and avoid SHOULD/MAY language as we don't test for it and, as such, implementers tend to ignore those statements for the most part.

Reply:
agree, but I will wait for addition pile on's or objections before merging.

Reply:
Seems more like the media type `can be used with`. We don't want the language to seem prescriptive -- such that you can't use this mime type to describe a payload unless it's being used with an external proof. For example, what if you're sending an insecure credential to a service to be signed -- where an embedded proof will be attached? You should be able to use this mime type to describe the payload.

Reply:
> We don't want the language to seem prescriptive -- such that you can't use this mime type to describe a payload unless it's being used with an external proof

Agreed, we think this media type has use as a `content-type` response from web servers, in that case there would be no `external proof`.

> For example, what if you're sending an insecure credential to a service to be signed -- where an embedded proof will be attached? You should be able to use this mime type to describe the payload.

exactly.

Reply:
We SHOULD be testing for SHOULD (and optimally for MAY) because the results of those tests may be differentiators for deployers choosing implementations to deploy.  Failing either SHOULD or MAY does not get a "noncompliant" label, but it *does* get an X on a feature which you MAY want to be Y.

Reply:
@TallTed wrote:
> We SHOULD be testing for SHOULD

We've avoided doing this to date, mostly because of 1) lack of test suite authoring cycles, 2) lack of interest from implementers to test these features, 3) suggestions by other W3C WGs that SHOULD/MAY have ended up being mostly failed normative statements because implementers can always come up with a good reason not to follow the guidance and so, effectively, SHOULD/MAYs are not treated in the way spec authors intend them to work.

The most compelling reasons are 1 and 2, IMHO, having been one of the people responsible for putting together the test suites. I don't have the spare cycles to write tests that hardly anyone is going to test against... and for those that don't test, there are some that'll complain that the tests exist at all (because their customers ask them -- "Why don't you run against these optional tests?" -- and yes, I know that's exactly the reason to write the tests in the first place). It's like maintaining the DID Spec Registries -- it's a completely thankless job, sucks up time that could be spent elsewhere, and results in a non-trivial amount of harassment to those that volunteer to do the work.

All that said, we did some variation of it for VCDM 1.0 and 1.1, there are optional features that we test, but those statements were more along the lines of "if you use feature X, then you *MUST* do Y"... not "if you do use feature X, then you *SHOULD* do Y". 

Food for thought, just highlighting the difficulties in that approach.

Reply:
@OR13 wrote:
> On reflection, can be used with seems like an improvement, @msporny mind updating your suggestion?

Done.

Reply:
> We've avoided doing this to date

Yeah, I'm aware of the need for cycles which are in short supply. This is one of the few arenas that make me want to be a coder ... and also grateful that I'm not.

This is also one of the more visible parts of "make the user's life easier than the implementer's, which should be easier than the spec developer's".  We really SHOULD NOT not-test SHOULDs and MAYs, because those tests can be key to the implementer's and/or the user's success, especially but not only with edge and/or corner case deployments/uses.

Reply:
Resolving this thread based on the merged suggestion here: https://github.com/w3c/vc-data-model/pull/1014/commits/a99fe01b917e5ce160cf78031e2d6e12546d6630